include/irr/asset/filters/dithering/CDither.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2020 - AnastaZIuk
+// This file is part of the "IrrlichtBAW" engine.
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#ifndef __IRR_C_DITHER_H_INCLUDED__
+#define __IRR_C_DITHER_H_INCLUDED__
+
+#include "../include/irr/asset/filters/dithering/IDither.h"
+
+namespace irr
+{
+	namespace asset
+	{
+		template<class CRTP>
+		class CDither : public IDither
+		{
+			public:
+				CDither() {}
+				virtual ~CDither() {}
+				
+				float pGet(const core::vectorSIMDu32& pixelCoord) override
+				{
+					return CRTP::get(pixelCoord);
+				}
+
+				class CState : public IDither::IState
+				{
+					public:
+						CState() {}
+						virtual ~CState() {}
+
+						ImageAsset imageAsset; // do we want to only handle ICPUImage or should I distinguish it to ICPUImageView?
```

Comment:
we should make a decision about it

Reply:
dont hold this here, remember `CDither` is a base class for many things like:
- CWhiteNoiseDither
- CBayerMatrixDither
- CPrecomputedDither

Reply:
not an issue now